src/program/program.service.ts

```diff
@@ -1083,11 +1083,10 @@ export class ProgramService {
 			data: {
 				...(type && { type }),
 				...(link && { link }),
-				parent: parent ? { connect: { id: parent } } : undefined
+				parent: parent ? { connect: { id: parent } } : undefined,
+				primary: primary
 			}
 		});
-
-		return this.prisma.content.update(data);
```

Comment:
You removed the return statement? 
I'm not sure if you didn't even test your changes or you accidentally deleted it but this breaks the service regardless.